### ISSUE TITLE
 Hello, I'd like to ask you about some experimental results.

### ISSUE THREAD

**Issue by Junyanzhang** [penghu-cs/UCCH#9]
****For COCO in 128 bit**   in paper TABLE 2**: ，Img2Txt:  0.665  Txt2Img: 0.677 **For Fig. 9** Img2Txt:  0.83  Txt2Img: 0.83  on the validation sets
However, I used your dataset for the experimental results as follows：
Epoch: 18 / 20
 [========================================== 457/457 
Evaluation
Img2Txt: 0.629952 	 Txt2Img: 0.628445 	 Avg: 0.629199
Saving..

Epoch: 19 / 20
 [========================================= 457/457 
Evaluation
Img2Txt: 0.631842 	 Txt2Img: 0.630144 	 Avg: 0.630993
Saving..
Test
**Img2Txt: 0.786291 	 Txt2Img: 0.805052 	 Avg: 0.795671**
Saving..

****For IAPR TC-12 in 128 bit**   in paper TABLE 1**，Img2Txt:  0.508  Txt2Img: 0.508  **For Fig. 9** Img2Txt:  0.505  Txt2Img: 0.505  on the validation sets
However,  I used your dataset for the experimental results as follows：
Epoch: 18 / 20
 70/70 
Evaluation
Img2Txt: 0.496356 	 Txt2Img: 0.496224 	 Avg: 0.49629

Epoch: 19 / 20
 70/70 
Evaluation
Img2Txt: 0.496524 	 Txt2Img: 0.496117 	 Avg: 0.49632
**Test
Img2Txt: 0.629495 	 Txt2Img: 0.630433 	 Avg: 0.629964**

Saving..

The difference is too significant, whether it's the test set or the validation set. Could it be that there's an issue with the way I ran it? Looking forward to your response. Thank you once again.

**Comment by penghu-cs** [comment]
Thanks a lot. For Fig. 9, we randomly select some samples from the retrieval set as a validation set, which is different from the validation set in the released code. Therefore, your results will differ slightly from those in Fig. 9. Moreover, the retrieval and query sets are the same as our experiments. Your results are much higher than ours. I think this difference may be due to different parameters or some careless modification (such as modifying the dataloader).

**Comment by JudyZhu45** [comment]
check func: fx_calc_map_multilabel_k 
order = ord[i].reshape(-1)[:k]
delete k or it will extract the best 2000 match from all retrieval set to compute MAP scores ,which will contribute to the higher result
